corehq/apps/analytics/static/analytics/js/kissmetrics.js

```diff
@@ -0,0 +1,152 @@
+    });
+
+    /**
+     * Identifies the current user
+     * @param {string} identity - A unique ID to identify the session.
+     */
+    var identify = function (identity) {
+        logger.debug.log(arguments, 'Identify');
+        _kmqPushCommand('identify', identity);
+    };
+
+    /**
+     * Sets traits for the current user
+     * @param {object} traits - an object of traits
+     * @param {function} callbackFn - (optional) callback function
+     * @param {integer} timeout - (optional) timeout in milliseconds
+     */
+    var identifyTraits = function (traits, callbackFn, timeout) {
+        logger.debug.log(logger.fmt.labelArgs(["Traits", "Callback Function", "Timeout"], arguments), 'Identify Traits (Set)');
+        callbackFn = _utils.createSafeCallback(callbackFn, timeout);
+        _kmqPushCommand('set', traits, callbackFn);
+    };
+
+    /**
+     * Records an event and its properties
+     * @param {string} name - Name of event to be tracked
+     * @param {object} properties - (optional) Properties related to the event being tracked
+     * @param {function} callbackFn - (optional) Function to be called after the event is tracked.
+     * @param {integer} timeout - (optional) Timeout for safe callback
+     */
+    var trackEvent = function (name, properties, callbackFn, timeout) {
+        logger.debug.log(arguments, 'RECORD EVENT');
+        callbackFn = _utils.createSafeCallback(callbackFn, timeout);
+        _kmqPushCommand('record', properties, callbackFn, name);
+    };
+
+    /**
+     * Tags an HTML element to record an event when its clicked
+     * @param {string} selector - The ID or class of the element to track.
+     * @param {string} name - The name of the event to record.
+     * @param {object} properties - optional Properties related to the event being recorded.
+     */
+    var trackClick = function (selector, name, properties) {
+        logger.debug.log(logger.fmt.labelArgs(["Selector", "Name", "Properties"], arguments), 'Track Click');
+        _kmqPushCommand('trackClick', properties, undefined, name);
+    };
+
+    /**
+     * Tags a link that takes someone to another domain and provides enough time to record an event when the link is clicked, before being redirected.
+     * @param {string} selector - The ID or class of the element to track.
+     * @param {string} name - The name of the event to record.
+     * @param {object} properties - optional Properties related to the event being recorded.
+     */
+    var trackOutboundLink = function (selector, name, properties) {
+        logger.debug.log(logger.fmt.labelArgs(["Selector", "Name", "Properties"], arguments), 'Track Click on Outbound Link');
+        _kmqPushCommand('trackClickOnOutboundLink', properties, undefined, name);
+    };
+
+    return {
+        logger: logger,
+        identify: identify,
+        identifyTraits: identifyTraits,
+        track: {
+            event: trackEvent,
+            click: trackClick,
```

Comment:
What would you think of renaming this to something like `internalClick`? I worry that people won't realize `outboundLink` exists.

Reply:
great point